### PR TITLE
Update documentation: complete client docs for honoring proxy env vars

### PR DESCRIPTION
**Pull request by celloni**
## What do these changes do?

* Update proxy support docs to honor `no_proxy` env
* Complete `trust_env` parameter description to honor `wss_proxy`, `ws_proxy` or `no_proxy` env

## Are there changes in behavior for the user?

No

## Related issue number

Update docs for: https://github.com/aio-libs/aiohttp/issues/4431

## Checklist

- [ ] I think the code is well written
- [ ] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [ ] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names.
- [ ] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` for example (588.bugfix)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

**Comment by Dreamsorcerer** [comment]
I'm not sure that's accurate, but I'd rather we linked to the documentation upstream here.
We just call https://docs.python.org/3/library/urllib.request.html#urllib.request.getproxies
Then the code filters the results to get proxies for these schemes: `("http", "https", "ws", "wss")`

**Comment by celloni** [comment]
Thanks a lot for the reply. Indeed this is more accurate. I tried to address the comment by linking to upstream in `client_advanced#proxy-support` and simplifying the rest by referencing to `client_advanced#proxy-support`

File: docs/client_advanced.rst [hunk 1]
```diff
@@ -592,14 +594,18 @@ Authentication credentials can be passed in proxy URL::
...
 
    async with aiohttp.ClientSession(trust_env=True) as session:
        async with session.get("http://python.org") as resp:
            print(resp.status)
 
+.. note::
+    ``aiohttp`` uses `urllib.request.getproxies <https://docs.python.org/3/library/urllib.request.html#urllib.request.getproxies>`_
+    for reading the proxy configuration and applies them for the schemas ``http``, ``https``, ``ws`` and ``wss``
+
+    Hosts defined in ``no_proxy`` will bypass the proxy.
```
Comment (Dreamsorcerer) [inline docs/client_advanced.rst:607]: If this is true, it should probably be documented in the Python docs. I couldn't find anything about no_proxy in our code or in the Python documentation last time I looked.

File: docs/client_advanced.rst [hunk 2]
```diff
@@ -592,14 +594,18 @@ Authentication credentials can be passed in proxy URL::
...
-*HTTP_PROXY*, *HTTPS_PROXY*, *WS_PROXY* or *WSS_PROXY* *environment
-variables* (all are case insensitive)::
+constructor.::
 
    async with aiohttp.ClientSession(trust_env=True) as session:
        async with session.get("http://python.org") as resp:
            print(resp.status)
 
+.. note::
+    ``aiohttp`` uses `urllib.request.getproxies <https://docs.python.org/3/library/urllib.request.html#urllib.request.getproxies>`_
```
Comment (Dreamsorcerer) [inline docs/client_advanced.rst]:
```suggestion
    aiohttp uses :func:`urllib.request.getproxies`
```
Intersphinx should link it automatically (and will error if it fails to link it).

File: docs/client_advanced.rst [hunk 3]
```diff
@@ -592,14 +594,18 @@ Authentication credentials can be passed in proxy URL::
...
+    for reading the proxy configuration and applies them for the schemas ``http``, ``https``, ``ws`` and ``wss``
```
Comment (Dreamsorcerer) [inline docs/client_advanced.rst]:
```suggestion
    for reading the proxy configuration (e.g. from the *HTTP_PROXY* etc. environment variables) and applies them for the *HTTP*, *HTTPS*, *WS* and *WSS* schemes.
```

File: docs/client_advanced.rst (hunk 1 above)
Comment (celloni) [reply docs/client_advanced.rst:607]: `getproxies` is reading `.*_proxy` from the env var and therefore as well `no_proxy`
```
// getproxies ouptut
{'no': 'localhost, 'http': 'some-host.com'}
```
The code uses then the helper function [urllib.request.proxy_bypass](https://github.com/python/cpython/blob/3.11/Lib/urllib/request.py#L2643) at this [postion](https://github.com/aio-libs/aiohttp/blob/master/aiohttp/helpers.py#L303) to check if the proxy shall be bypassed for a given host.
Unfortunaltey `urllib.request.proxy_bypass` isn't documneted like `getproxies` in the python docs.

**Comment by Dreamsorcerer** [comment]
If you could handle the backport, as above, that would be great.

**Comment by celloni** [comment]
Created the backport here https://github.com/aio-libs/aiohttp/pull/7256

**Pull request merged**